lib/segment/src/vector_storage/dense/appendable_dense_vector_storage.rs

```diff
@@ -329,3 +348,91 @@ pub fn open_appendable_in_ram_vector_storage_impl<T: PrimitiveVectorElement>(
         _phantom: Default::default(),
     })
 }
+
+/// Find files related to this dense vector storage
+pub(crate) fn find_storage_files(vector_storage_path: &Path) -> OperationResult<Vec<PathBuf>> {
+    let vectors_path = vector_storage_path.join(VECTORS_DIR_PATH);
+    let deleted_path = vector_storage_path.join(DELETED_DIR_PATH);
+
+    let mut files = vec![];
+    files.extend(list_files(&vectors_path)?);
+    files.extend(list_files(&deleted_path)?);
+    Ok(files)
+}
+
+/// List all files in the given directory recursively.
+fn list_files(dir: &Path) -> OperationResult<Vec<PathBuf>> {
```

Comment:
why not use the already present [`walkdir`](https://docs.rs/walkdir/latest/walkdir/)?

Reply:
Ended up doing it as part of (<https://github.com/qdrant/qdrant/pull/6607>): https://github.com/qdrant/qdrant/pull/6607/commits/5d02d65d966e78ea1174fc593a90376bdf875a77

The difference isn't huge, but since we already had walkdir I consider it to be better.